python/tskit/tables.py

```diff
@@ -3862,6 +3862,56 @@ def trim(self, record_provenance=True):
                 record=json.dumps(provenance.get_provenance_dict(parameters))
             )
 
+    def decapitate(self, time, *, record_provenance=True):
+        """
+        Delete all edge topology and mutational information older than the
+        specified time from this set of tables.
+
+        Removes all edges in which the time of the child is >= the specified
+        time ``t``, and breaks edges that intersect with ``t``. For each edge
+        intersecting with ``t`` we create a new node with time equal to ``t``,
+        and set the parent of the edge to this new node. The node table
+        is not altered in any other way. Newly added nodes have empty metadata,
+        a ``flags`` value of 0, and NULL ``population`` and ``individual``
+        references.
+
+        .. warning::
+            The empty metadata values for newly added nodes may not be compatible
+            with the node table's metadata schema! The current behaviour may
+            change in future versions to better accomodate metadata schemas.
```

Comment:
hah you thought of this already =)

Reply:
That list of example tree sequences doesn't let you get away with much!

I don't really know how to deal with this properly tbh, other than providing options to fill out the metadata field for the new nodes. This still won't guarantee people get it right, of course. We just have to make a note and move on I think, unless @benjeffery has any better ideas?

Reply:
I agree this is a bigger issue, but it does mean that we can't simply decapitate SLiM tree sequences.

I could imagine strongly suggesting that all schema that don't allow empty metadata provide defaults, and sticking in defaults in such situations (not sure the clean way to do that).